lib/semian/lru_cache.rb

```diff
@@ -0,0 +1,54 @@
+class LRUCache
+  extend Forwardable
+  def_delegators :@table, :size, :count
+  def_delegators :@table, :empty?
+  def_delegators :@table, :keys
+  def_delegators :@table, :clear
+
+  attr_accessor :table, :minimum_time_in_queue
+  def initialize(minimum_time_in_queue = 5)
+    @minimum_time_in_queue = minimum_time_in_queue * 60
+    @table = {}
+  end
+
+  def set(key, value)
+    @table.delete(key)
+    @table[key] = value
+    clear_unused_resources
+    value
+  end
+
+  def get(key)
+    if found = @table.delete(key)
+      @table[key] = found
+    end
+  end
+
+  def []=(key, value)
+    set(key, value)
+  end
+
+  def [](key)
+    get(key)
+  end
+
+  def delete(key)
+    @table.delete(key)
+  end
+
+  private
+
+  def clear_unused_resources
+    cleaned_elements = 0
+    @table.each do |key, value|
+      break if cleaned_elements >= 2
```

Comment:
Not completely sure how it'd fare in the real world, your worst case here is still to traverse _a lot_ of resources 🤷‍♂️  What kinds of benchmarks have you run? Again, because this is only run on `set`, it doesn't seem too bad, I'm just not sure I understand what logic lead you to this decision.

Reply:
No test fails if I remove it or change the number though.